proto/api/v1beta1/event.proto

```diff
@@ -2,38 +2,121 @@ syntax = "proto3";
 
 package ensign.v1beta1;
 
+import "region/v1beta1/region.proto";
 import "mimetype/v1beta1/mimetype.proto";
 import "google/protobuf/timestamp.proto";
 
+// EventWrapper is a header wrapper for processing events by the Ensign server. It is
+// intended to reduce the processing time of events to increase throughput on the server
+// even though it will require more work from the SDKs to serialize events. However,
+// this will happen on the client-side making Ensign nodes much more parallel. The only
+// data in the EventWrapper should be data that is required for the Ensign node to
+// process events.
+message EventWrapper {
```

Comment:
So for right now, the SDKs mostly just have to set the `topic_id` and serialize the `Event` into the `event` field?

Reply:
That's correct -- and possibly the `key` for sharding topics. I was definitely thinking server-side for most of these fields.

Reply:
Also note that I made the topic_id the ULID bytes instead of ULID string to make the event wrapper more compact.